### PR TITLE
Handle `multiSend` relay limiting

### PR DESCRIPTION
**Pull request by iamacook**
This [ports the limiting](https://github.com/safe-global/safe-gelato-relay-service/blob/main/src/routes/relay/entities/schema/sponsored-call.schema.ts#L44-L45) of `multiSend` relays based on the extraction of a "valid" Safe address:

1. Calldata is checked to be `multiSend`.
2. Individual transactions are decoded.
3. Each is checked to be a "valid" `execTransaction`.
4. Each is checked to have the same recipient.
5. The recipient is returned.

### CODE REVIEW

File: src/domain/relay/limit-addresses.mapper.ts [hunk 1]
```diff
@@ -69,4 +75,38 @@ export class LimitAddressesMapper {
     const isCancellation = execTransaction.data === '0x';
     return isCancellation || this.safeContract.isCall(execTransaction.data);
   }
+
+  private isMultiSend(data: Hex): boolean {
+    return this.multiSendDecoder.isFunctionCall({
+      functionName: 'multiSend',
+      data,
+    });
```
[the remark this answers is not in the document]
Comment (iamacook) [reply src/domain/relay/limit-addresses.mapper.ts]: I migrated `isFunctionCall` to the `AbiDecoder` and moved this into the `MultiSendDecoder` in df4649dd2c309bdcba327abe6b7ecce572de113e.

I've added a note to improve this: I think the ideal solution would be autogenerated helpers according to the ABI.

File: src/domain/relay/limit-addresses.mapper.ts [hunk 2]
```diff
@@ -69,4 +75,38 @@ export class LimitAddressesMapper {
...
+      return this.isValidExecTransactionCall(transaction.to, transaction.data);
+    });
+
+    if (!isEveryValid) {
+      throw Error('Invalid MultiSend transactions');
+    }
+
+    const firstRecipient = transactions[0].to;
+
+    // Every transaction is 'self' (the Safe)
```
[the remark this answers is not in the document]
Comment (iamacook) [reply src/domain/relay/limit-addresses.mapper.ts]: This is a leftover from comparing code with the relay service. I removed the comment in df4649dd2c309bdcba327abe6b7ecce572de113e.

File: src/domain/alerts/contracts/abi-decoder.helper.ts [hunk 3]
```diff
@@ -38,4 +38,18 @@ export abstract class AbiDecoder<TAbi extends Abi | readonly unknown[]> {
   ) {
     return _decodeFunctionData({ ...args, abi: this.abi });
   }
+
+  isFunctionCall(args: {
+    functionName: ContractEventName<TAbi>;
```
[the remark this answers is not in the document]
Comment (iamacook) [reply src/domain/alerts/contracts/abi-decoder.helper.ts:44]: I changed it to `ContractFunctionName` in df4649dd2c309bdcba327abe6b7ecce572de113e.

File: src/domain/relay/limit-addresses.mapper.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fmrsabino) [reply src/domain/relay/limit-addresses.mapper.ts]: As an initial step we can write the functions ourselves to avoid having a generation step initially.

**Pull request merged**